artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQConnectionFactory.java

```diff
@@ -96,9 +94,6 @@ public class ActiveMQConnectionFactory extends JNDIStorable implements Connectio
 
    private boolean cacheDestinations;
 
-   // keeping this field for serialization compatibility only. do not use it
-   private boolean finalizeChecks;
```

Comment:
Similar pondering to https://github.com/apache/activemq-artemis/pull/5477#discussion_r1933742419, with difference that I can see that the relation to JNDIBaseStorable here probably means this really is actually needed for compatibility and should stay.